src/Components/ShortcutKeyModal/SearchModal.js

```diff
@@ -0,0 +1,142 @@
+import React, { useRef, useEffect, useState, useCallback, useMemo } from 'react';
+import './SearchModal.css';
+import { officerOrAdminRoutes, signedOutRoutes, memberRoutes, notAuthenticatedRoutes } from '../../Routes';
+import { membershipState } from '../../Enums';
+import { useUser } from '../context/UserContext';
+
+export default function SearchModal({ appProps }) {
+  const [open, setOpen] = useState(false);
+  const inputRef = useRef(null);
+  const [keyword, setKeyword] = useState('');
+  const [suggestions, setSuggestions] = useState([]);
+  const [selectItem, setSelectItem] = useState(0);
+  const { user } = useUser();
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const routes = useMemo(() => {
+    if (user.accessLevel === membershipState.MEMBER) return [...memberRoutes, ...signedOutRoutes];
+    if (user.accessLevel >= membershipState.OFFICER) return [...officerOrAdminRoutes, ...signedOutRoutes];
+    if (!appProps.authenticated) return [...notAuthenticatedRoutes, ...signedOutRoutes];
+    return [...signedOutRoutes];
+  }, [user.accessLevel]);
+
+  function handleChanges(e) {
+    setKeyword(e.target.value);
+    setSelectItem(0);
+  }
+
+  function getSuggestions() {
+    if (suggestions.length === 0) return <></>;
+
+    return (
+      <ul className='suggestion-list'>
+        {suggestions.map((r, index) => (
+          <li
+            key={index}
```

Comment:
```suggestion
        {suggestions.map((r) => (
          <li
            key={r.path}
```

https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66

Reply:
Use r.path as key and still keep index to keep track of the cursor.